File: I. Beginner_Projects/3_Inches To Meter Converter/app/src/main/java/com/abrarhamim/inchestometerconverter/MainActivity.java
package com.abrarhamim.inchestometerconverter;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import java.text.DecimalFormat;


public class MainActivity extends AppCompatActivity {

    private EditText inchesEditText;
    private Button buttonConvert;
    private TextView meterResultText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findView();
        setUpButtonClickConverter();
    }

    private void findView() {
        inchesEditText = findViewById(R.id.edit_text_inches);
        buttonConvert = findViewById(R.id.button_convert);
        meterResultText = findViewById(R.id.text_view_meter_result);
    }

    private void setUpButtonClickConverter() {
        buttonConvert.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String inchesString = inchesEditText.getText().toString();


                if (inchesString.isEmpty()) {
                    Toast.makeText(MainActivity.this, "Please input inches", Toast.LENGTH_LONG).show();
                } else {
                    double result = calculateHeight(inchesString);
                    displayResult(result);
                }
            }
        });
    }

    private double calculateHeight(String inchesString) {
        int inches = Integer.parseInt(inchesString);
        return inches * 0.0254;
    }

    private void displayResult(double result) {
        DecimalFormat myFormatter = new DecimalFormat("0.00");
        String resultString = myFormatter.format(result);
        meterResultText.setText(resultString + " meters");
    }

}

File: I. Beginner_Projects/10_Activity Lifecycle Playground/app/src/main/java/com/abrar/activitylifecycleplayground/MainActivity.kt
package com.abrar.activitylifecycleplayground

import android.os.Bundle
import android.widget.Toast
import androidx.activity.addCallback
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.abrar.activitylifecycleplayground.databinding.ActivityMainBinding
import java.io.File
import java.util.Timer

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private var seconds = 0
    private lateinit var timer: Timer

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        binding.buttonExit.setOnClickListener { showDialog() }
        binding.buttonSave.setOnClickListener { saveMessage() }
        binding.buttonShowFragment.setOnClickListener { showFragment() }
        binding.buttonRemoveFragment.setOnClickListener { removeFragment() }
        val callback = onBackPressedDispatcher.addCallback { showDialog() }

        binding.textViewSavedMessage.text = savedInstanceState?.getString("savedMessage")
    }

    private fun showFragment() {

    }

    private fun removeFragment() {

    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        val savedTextViewMessage = binding.textViewSavedMessage.text.toString()
        outState.putString("savedMessage", savedTextViewMessage)
    }

    private fun saveMessage() {
        val userMessage = binding.editTextMessage.text
        File(filesDir, "user message.txt").writeText(userMessage.toString())
        binding.textViewSavedMessage.text =
            "Your message has been saved!\n\nMessage Preview:\n\n$userMessage"
        binding.editTextMessage.setText("")
    }

    private fun showDialog() {
        AlertDialog.Builder(this)
            .setTitle("Warning!")
//            .setMessage("You are about to leave the app. Are you sure you want to exit?")
            .setView(R.layout.dialog_warning)
            .setPositiveButton("Yes") { _, _ -> finish() }
            .setNegativeButton("No") { dialog, _ -> dialog.dismiss() }
            .setNeutralButton("More info") { dialog, _ ->
                Toast.makeText(
                    this,
                    "This is where the more info screen could be!",
                    Toast.LENGTH_LONG
                ).show()
                dialog.dismiss()
            }
            .show()
    }
}